### PR TITLE
Ensure logAnalyticsWorkspaceId parameter is present in Aspire manifest.

### LINKED ISSUE #3133: AppInsights resource does not define logAnalyticsWorkspaceId as a parameter
This stops azd from using the one associated with the container apps environment. Looks like a regression from the CDK work.

### PR DESCRIPTION
**Pull request by mitchdenny**
Fixes #3133 and #3273

To fix this I've added an argument to the `AddAzureApplicationInsights(...)` extension methods that allows it to take an `IResourceBuilder<AzureLogAnalyticsWorkspaceResource>` and uses a reference to that resource to populate the `logAnalyticsWorkspaceId` parameter in the manifest.

If this value is not passed in we still ensure that the parameter is in the manifest so that it can be supplied by AZD.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/3296)

### CODE REVIEW

File: playground/bicep/BicepSample.AppHost/aspire-manifest.json [hunk 1]
```diff
@@ -23,7 +23,7 @@
     },
     "test0": {
       "type": "azure.bicep.v0",
-      "path": "../../../../../../Users/davifowl/AppData/Local/Temp/tmppj1k21.tmp.bicep"
+      "path": "../../../../../Users/midenn/AppData/Local/Temp/tmpvqsa5k.tmp.bicep"
```
Comment (davidfowl) [inline playground/bicep/BicepSample.AppHost/aspire-manifest.json:26]: Yikes 😄

Comment (mitchdenny) [reply playground/bicep/BicepSample.AppHost/aspire-manifest.json:26]: Yep. I've got an issue for this I think: https://github.com/dotnet/aspire/issues/3273

Comment (mitchdenny) [reply playground/bicep/BicepSample.AppHost/aspire-manifest.json]: OK I integrated the change to fix the path in thsi PR too.

**Comment by mitchdenny** [comment]
@joperezr we will want to back port this one to preview 5.

**Review by davidfowl — APPROVED** [review]

File: src/Aspire.Hosting.Azure/AzureBicepResource.cs [hunk 2]
```diff
@@ -73,8 +73,12 @@ public virtual BicepTemplateFile GetBicepTemplateFile(string? directory = null,
         {
             isTempFile = directory is null;
 
+            //path = TempDirectory is null
+            //    ? Path.GetTempFileName() + ".bicep"
+            //    : Path.Combine(TempDirectory, $"{Name.ToLowerInvariant()}.bicep");
```
Comment (davidfowl) [inline src/Aspire.Hosting.Azure/AzureBicepResource.cs:78]: Remove dead code
```suggestion
```

File: src/Aspire.Hosting.Azure/AzureBicepResource.cs [hunk 3]
```diff
@@ -73,8 +73,12 @@ public virtual BicepTemplateFile GetBicepTemplateFile(string? directory = null,
...
+
```
Comment (davidfowl) [inline src/Aspire.Hosting.Azure/AzureBicepResource.cs:79]: Dead code
```suggestion
```

**Pull request merged**